browser/mapping_test.go

```diff
@@ -239,3 +235,360 @@ func isCustomMapping(customMappings map[string]string, typ, method string) (stri
+}
+
+// requestAPI is the interface of an HTTP request.
+type requestAPI interface {
+	AllHeaders() map[string]string
+	Failure() goja.Value
+	Frame() *common.Frame
+	HeaderValue(string) goja.Value
+	Headers() map[string]string
+	HeadersArray() []common.HTTPHeader
+	IsNavigationRequest() bool
+	Method() string
+	PostData() string
+	PostDataBuffer() goja.ArrayBuffer
+	PostDataJSON() string
+	RedirectedFrom() requestAPI
+	RedirectedTo() requestAPI
+	ResourceType() string
+	Response() *common.Response
+	Size() common.HTTPMessageSize
+	Timing() goja.Value
+	URL() string
+}
+
+// responseAPI is the interface of an HTTP response.
+type responseAPI interface {
+	AllHeaders() map[string]string
+	Body() goja.ArrayBuffer
+	Finished() bool
+	Frame() *common.Frame
+	HeaderValue(string) goja.Value
+	HeaderValues(string) []string
+	Headers() map[string]string
+	HeadersArray() []common.HTTPHeader
+	JSON() goja.Value
+	Ok() bool
+	Request() *common.Request
+	SecurityDetails() goja.Value
+	ServerAddr() goja.Value
+	Size() common.HTTPMessageSize
+	Status() int64
+	StatusText() string
+	URL() string
+}
+
+// locatorAPI represents a way to find element(s) on a page at any moment.
+type locatorAPI interface {
+	Click(opts goja.Value) error
+	Dblclick(opts goja.Value)
+	Check(opts goja.Value)
+	Uncheck(opts goja.Value)
+	IsChecked(opts goja.Value) bool
+	IsEditable(opts goja.Value) bool
+	IsEnabled(opts goja.Value) bool
+	IsDisabled(opts goja.Value) bool
+	IsVisible(opts goja.Value) bool
+	IsHidden(opts goja.Value) bool
+	Fill(value string, opts goja.Value)
+	Focus(opts goja.Value)
+	GetAttribute(name string, opts goja.Value) goja.Value
+	InnerHTML(opts goja.Value) string
+	InnerText(opts goja.Value) string
+	TextContent(opts goja.Value) string
+	InputValue(opts goja.Value) string
+	SelectOption(values goja.Value, opts goja.Value) []string
+	Press(key string, opts goja.Value)
+	Type(text string, opts goja.Value)
+	Hover(opts goja.Value)
+	Tap(opts goja.Value)
+	DispatchEvent(typ string, eventInit, opts goja.Value)
+	WaitFor(opts goja.Value)
+}
+
+// keyboardAPI is the interface of a keyboard input device.
+// TODO: map this to page.GetKeyboard(). Currently, the common.Keyboard type
+// mapping is not tested using this interface. We use the concrete type
+// without testing its exported methods.
+type keyboardAPI interface { //nolint: unused
+	Down(key string)
+	InsertText(char string)
+	Press(key string, opts goja.Value)
+	Type(text string, opts goja.Value)
+	Up(key string)
+}
+
+// touchscreenAPI is the interface of a touchscreen.
+// TODO: map this to page.GetTouchscreen(). Currently, the common.TouchscreenAPI type
+// mapping is not tested using this interface. We use the concrete type
+// without testing its exported methods.
+type touchscreenAPI interface { //nolint: unused
+	Tap(x float64, y float64)
+}
+
+// cdpSessionAPI is the interface of a raw CDP session.
+type cdpSessionAPI interface { //nolint: unused
```

Comment:
A comment for `cdpSessionAPI`, `videoAPI` and `routeAPI`. If we are not using them, why should we keep them?

Reply:
Sure, we can remove them: 43dfcab